web_sdk/sdk_rewriter.dart

```diff
@@ -123,6 +124,49 @@ String rewriteFile(String source, {required String filePath, required bool isUi,
   return source;
 }
 
+// Enforces a particular structure in engine.dart source code.
+//
+// Code in `engine.dart` must only be made of the library directive, exports,
+// and code comments. Imports are disallowed. Instead, the required imports are
+// added by this script during the rewrite.
+void _validateEngineSource(String engineDartPath, String engineDartCode) {
+  const List<String> expectedLines = <String>[
+    'library engine;',
```

Comment:
I was bitten before by having a different number of spaces. That's why I [added](https://github.com/flutter/engine/pull/25866) `\s+` to many regexes above. Maybe turn this into a list of regexes?

Another issue that I can think of is this line could have a trailing comment in the source code:
```dart
library engine; // some comment here.
```
Could also be fixed if we use a list of regexes.

If you think this is overkill now, you can just leave it as is. We can always fix it in the future if it becomes a problem.

Reply:
I'm guessing for this particular line it probably doesn't matter. But I agree, for code that's more malleable that could be annoying.